File: db4oj/test/src/db4ounit/extensions/fixtures/Db4oClientServer.java
/* Copyright (C) 2004 - 2006  db4objects Inc.  http://www.db4o.com */

package db4ounit.extensions.fixtures;

import java.io.File;

import com.db4o.Db4o;
import com.db4o.ObjectServer;
import com.db4o.ext.ExtObjectContainer;

import db4ounit.extensions.Db4oFixture;

public class Db4oClientServer implements Db4oFixture {
	private static final String HOST = "localhost";
	private static final String USERNAME = "db4o";
	private static final String PASSWORD = USERNAME;
	
	File _yap;
	private ObjectServer _server;
	private ExtObjectContainer _db;
	private int _port;
	
	public Db4oClientServer(String fileName, int port) {
		_yap = new File(fileName);
		_port = port;
	}

	public void clean() {
		_yap.delete();
	}

	public void close() throws Exception {
		_db.close();
		_server.close();
	}

	public ExtObjectContainer db() {
		return _db;
	}

	public void open() throws Exception {
		_server = Db4o.openServer(_yap.getCanonicalPath(), _port);
		_server.grantAccess(USERNAME, PASSWORD);
		_db = (ExtObjectContainer) Db4o.openClient(HOST, _port, USERNAME, PASSWORD);
	}
}

File: db4on/Db4oUnit.Extensions/src/Db4oUnit/Extensions/Fixtures/Db4oClientServer.cs
namespace Db4oUnit.Extensions.Fixtures
{
	public class Db4oClientServer : Db4oUnit.Extensions.Db4oFixture
	{
		private static readonly string HOST = "localhost";

		private static readonly string USERNAME = "db4o";

		private static readonly string PASSWORD = USERNAME;

		private com.db4o.ObjectServer _server;

		private readonly int _port;

		private readonly j4o.io.File _yap;

		public Db4oClientServer(string fileName, int port)
		{
			_yap = new j4o.io.File(fileName);
			_port = port;
		}

		public virtual void Close()
		{
			_server.Close();
		}

		public virtual void Open()
		{
			_server = com.db4o.Db4o.OpenServer(_yap.GetAbsolutePath(), _port);
			_server.GrantAccess(USERNAME, PASSWORD);
		}

		public virtual com.db4o.ext.ExtObjectContainer Db()
		{
			try
			{
				return com.db4o.Db4o.OpenClient(HOST, _port, USERNAME, PASSWORD).Ext();
			}
			catch (System.IO.IOException e)
			{
				j4o.lang.JavaSystem.PrintStackTrace(e);
				throw new Db4oUnit.TestException(e);
			}
		}

		public virtual void Clean()
		{
			_yap.Delete();
		}
	}
}
